include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -286,6 +374,37 @@ is_operational(const Lyt& lyt, const std::vector<TT>& spec, const is_operational
 
     return {p.run(), p.get_number_of_simulator_invocations()};
 }
+/**
+ * This function calculates the count of input combinations for which the SiDB-based logic, represented by the
+ * provided layout (`lyt`) and truth table specifications (`spec`), produces the correct output.
+ *
+ * @tparam Lyt Type of the cell-level layout.
+ * @tparam TT Type of the truth table.
+ * @param lyt The SiDB layout.
+ * @param spec Vector of truth table specifications.
+ * @param params Parameters to simualte if a input combination is operational.
+ * @return The count of operational input combinations.
+ */
+template <typename Lyt, typename TT>
+[[nodiscard]] std::set<uint64_t> number_of_operational_inputs(const Lyt& lyt, const std::vector<TT>& spec,
+                                                              const is_operational_params& params = {}) noexcept
```

Comment:
Given the new return type, this function seems misnamed now. It should rather be something like `operational_input_patterns`